controller/gatewayclass/controller_reconciler_utils.go

```diff
@@ -63,3 +67,61 @@ func getAcceptedCondition(ctx context.Context, cl client.Client, gwc *gatewayv1.
 
 	return &acceptedCondition, nil
 }
+
+// setSupportedFeatures sets the supported features in the gatewayClass status.
+// The set of supported features depends on the router flavor.
+func setSupportedFeatures(ctx context.Context, cl client.Client, gwc *gatewayv1.GatewayClass, gatewayConfig *operatorv1beta1.GatewayConfiguration) error {
+	flavor, err := getRouterFlavor(ctx, cl, gatewayConfig)
+	if err != nil {
+		return err
+	}
+	feats, err := gatewayapipkg.GetSupportedFeatures(flavor)
+	if err != nil {
+		return err
+	}
+	supportedFeatures := feats.UnsortedList()
+	slices.Sort(supportedFeatures)
+	gwc.Status.SupportedFeatures = lo.Map(supportedFeatures, func(f features.FeatureName, _ int) gatewayv1.SupportedFeature {
+		return gatewayv1.SupportedFeature{
+			Name: gatewayv1.FeatureName(f),
+		}
+	})
+
+	return nil
+}
+
+// getGatewayConfiguration returns the GatewayConfiguration referenced by the GatewayClass.
+func getGatewayConfiguration(ctx context.Context, cl client.Client, gwc *gatewayv1.GatewayClass) (*operatorv1beta1.GatewayConfiguration, error) {
+	gatewayConfig := operatorv1beta1.GatewayConfiguration{}
+
+	if gwc.Spec.ParametersRef == nil {
+		return nil, nil
+	}
+
+	err := cl.Get(ctx, client.ObjectKey{Name: gwc.Spec.ParametersRef.Name, Namespace: string(*gwc.Spec.ParametersRef.Namespace)}, &gatewayConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &gatewayConfig, nil
+}
+
+// getRouterFlavor returns the router flavor to be used by the GatewayClass. It is inferred by
+// the KONG_ROUTER_FLAVOR environment variable in the DataPlane proxy container.
+func getRouterFlavor(ctx context.Context, cl client.Client, gatewayConfig *operatorv1beta1.GatewayConfiguration) (consts.RouterFlavor, error) {
+	if gatewayConfig == nil ||
+		gatewayConfig.Spec.DataPlaneOptions == nil ||
+		gatewayConfig.Spec.DataPlaneOptions.Deployment.PodTemplateSpec == nil {
+		return consts.RouterFlavorExpressions, nil
```

Comment:
This means that the expressions router is the default right? Can we make this explicit? Perhaps adding a new `const` pointing to `RouterFlavorExpressions`. WDYT?

Reply:
yep, addressed in https://github.com/Kong/gateway-operator/pull/1010/commits/a34b25b67a77693827c5db919f64a168357bcd97#diff-3136fbc6fb2d5338f6dcb90a06fe1f41beb533ff718dbedf73b59b5c179dbea1